libcontainer/notify_v2_linux.go

```diff
@@ -40,7 +41,11 @@ func registerMemoryEventV2(cgDir, evName, cgEvName string) (<-chan struct{}, err
 
 		for {
 			n, err := unix.Read(fd, buffer[:])
```

Comment:
Overall LGTM, a small suggestion:
Maybe we can use `os.File` to read from data this fd.